add label F4 value help

christianguenter2 opened the pull request:
new value help feature for labels form field on pages:
- new online
- new offline
- local repo settings
![grafik](https://user-images.githubusercontent.com/17437789/204129214-6943cc5e-020d-4cd5-b43c-579e1b022109.png)

larshp commented:
would like to avoid the popups, but well, long story...

mbtools commented on src/ui/zcl_abapgit_gui_page_addofflin.clas.abap:
```diff
@@ -27,14 +27,14 @@ CLASS zcl_abapgit_gui_page_addofflin DEFINITION
         url            TYPE string VALUE 'url',
         package        TYPE string VALUE 'package',
         folder_logic   TYPE string VALUE 'folder_logic',
-        labels         TYPE string VALUE 'labels',
```
can we keep all `ids` on the same page instead of having them in multiple places?

mbtools commented on src/zif_abapgit_definitions.intf.abap:
```diff
@@ -548,4 +548,10 @@ INTERFACE zif_abapgit_definitions
+
+  CONSTANTS:
+    BEGIN OF c_id,
+      labels TYPE string VALUE 'labels',
```
keep `ids` on pages

christianguenter2 replied:
moved `labels` id to pages. Now we have some duplication within the `choose_labels` methods which I wanted to get rid. Therefore I moved the id to the global interface.

christianguenter2 replied:
@mbtools moved id to pages

mbtools reviewed (changes requested):
we'll get to HTML popups eventually :-)

sbcgua commented:
Nice feature. Certainly a convenient search helper. I have one concern. The `repo_labels` class is kind of deep backend. And now it knows about persistence (so it persistence depends on it and vice versa) and UI (same concern). Imho it would be better to keep it abstract and independent. Not critical, but I'd recommend:
1) move `get_label_list` to repo_srv persistence class (why not, it's repo meta data). 
    - Also maybe (maybe !!) unify it with `repo_over->collect_all_labels`. Because in essense they do similar thing ... though maybe it is not that easy, because a) collect_all_labels processes converted data b) it can work on specific list of repos (favs only) which does not fit well into repo_srv nature. So it is just an idea.
2) and move `choose_labels` somewhere to UI. Maybe to popups ? This will potentially simplify further conversion to html (when and if it will be done ;) because the popups will be all together

christianguenter2 commented:
@sbcgua very valid concerns.

1. Decided to move `get_label_list` to `zcl_abapgit_repo_srv`
2. Moved `choose_labels` back to pages. Main logic is already in `popup` class. But now we have a minor duplication in all three page classes but might be ok. That's why I moved it to 'labels' class initially.

mbtools commented on src/ui/zcl_abapgit_popups.clas.abap:
```diff
@@ -1278,4 +1298,84 @@ CLASS zcl_abapgit_popups IMPLEMENTATION.
+        it_list               = lt_all_labels
+        iv_selection_mode     = if_salv_c_selection_mode=>multiple
+        it_columns_to_display = lt_columns_to_display
+        it_preselected_rows   = lt_preselected_rows
```
add 
```
        iv_start_column       = 15
        iv_end_column         = 55
```

christianguenter2 replied:
Thx. Fixed.

mbtools reviewed (changes requested):
Works well. Popup is just a bit big, maybe. Better:

![image](https://user-images.githubusercontent.com/59966492/204657587-3fb49114-ba68-4fae-84fa-7291fe7ab072.png)

sbcgua commented:
`get_label_list`, `choose_labels` :+1:
Looks good.

`choose_labels` duplication: yeah ... not sure what to do with that either. It is not a chunk, it is not base method. In fact it is a one-liner, grown to a separate method for better semantics purposes (which is good). So I think it is ok. :)

mbtools reviewed (approved):
LGTM 👍
